src/server/session.ts

```diff
@@ -691,9 +692,11 @@ namespace ts.server {
         private suppressDiagnosticEvents?: boolean;
         private eventHandler: ProjectServiceEventHandler | undefined;
         private readonly noGetErrOnBackgroundUpdate?: boolean;
+        private fshost?: FileServerHost;
 
         constructor(opts: SessionOptions) {
             this.host = opts.host;
+            this.fshost = opts.fshost;

```

Comment:
```suggestion
            this.fshost = opts.fshost; || this.host;
```

With this you can get rid of all the fsHost || host from session too..